site/src/components/PaginationWidget/PaginationWidget.tsx

```diff
@@ -0,0 +1,158 @@
+import Button from "@material-ui/core/Button"
+import { makeStyles } from "@material-ui/core/styles"
+import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft"
+import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight"
+import { CSSProperties } from "react"
+
+export type PaginationWidgetProps = {
+  prevLabel: string
+  nextLabel: string
+  onPrevClick: () => void
+  onNextClick: () => void
+  onPageClick?: (page: number) => void
+  numRecordsPerPage?: number
+  numRecords?: number
+  activePage?: number
+  containerStyle?: CSSProperties
+}
+
+/**
+ * Generates a ranged array with an option to step over values.
+ * Shamelessly stolen from:
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#sequence_generator_range
+ */
+const range = (start: number, stop: number, step = 1) =>
+  Array.from({ length: (stop - start) / step + 1 }, (_, i) => start + i * step)
+
+const DEFAULT_RECORDS_PER_PAGE = 25
+// Number of pages to the left or right of the current page selection.
+const PAGE_NEIGHBORS = 1
+// Number of pages displayed for cases where there are multiple ellipsis showing. This can be
+// thought of as the minimum number of page numbers to display when multiple ellipsis are showing.
+const PAGES_TO_DISPLAY = PAGE_NEIGHBORS * 2 + 3
+// Total page blocks(page numbers or ellipsis) displayed, including the maximum number of ellipsis (2).
+// This gives us maximum number of 7 page blocks to be displayed when the page neighbors value is 1.
+const NUM_PAGE_BLOCKS = PAGES_TO_DISPLAY + 2
+
+export const PaginationWidget = ({
+  prevLabel,
+  nextLabel,
+  onPrevClick,
+  onNextClick,
+  onPageClick,
+  numRecords,
+  numRecordsPerPage = DEFAULT_RECORDS_PER_PAGE,
+  activePage = 1,
+  containerStyle,
+}: PaginationWidgetProps): JSX.Element | null => {
+  const numPages = numRecords ? Math.ceil(numRecords / numRecordsPerPage) : 0
+  const firstPageActive = activePage === 1 && numPages !== 0
+  const lastPageActive = activePage === numPages && numPages !== 0
+
+  const styles = useStyles()
+
+  const buildPagedList = () => {
```

Comment:
I like to pull pure functions out of components as much as possible, maybe that can be done here? Then they can be unit tested easily, although I know you've tested the component, so it's up to you.

Reply:
Sure, I can take a stab at that!

Reply:
I have a diff opinion, I think it is easier to manage/find code by keeping them close to where they are used, and it looks like this function only will be used on this component so I would keep it in the component file but it is up to you.

Reply:
@presleyp @BrunoQuaresma I compromised here and added some unit tests but didn't separate the function from its component file so it should still be easy to find.

Reply:
Great! I think separating the function from the component is the important part.